### PR TITLE
Add abstract class for MustFightBattle & NonFightingBattle to consoli…

### PR DESCRIPTION
**Pull request by simon33-2**
…date duplicated code

Certain methods are common between NonFightingBattle and MustFightBattle so add a class for these to inherit from and remove duplicated code.

### CODE REVIEW

**Comment by ron-murhammer** [comment]
@simon33-2 Will this break save game compatibility since delegates are part of the game data?

**Comment by simon33-2** [comment]
I don't believe so. I haven't had any problems loading save games created with previous versions or vice versa.

Would an abstract class be included in the game data?

**Comment by DanVanAtta** [comment]
The serialization uid may be at play. There used to be at least 3 hidden compatibility requirements in the game. We recently removed the reflection of XML tags to objects. Another is save games, the third are games transmitted during network play. This last case is perhaps worth checking, BattleDelegates are not saved, but they might be transmitted.

A way to check is to host a local game, switch to master and then join that game on a second instance. If you can get to a combat and roll some dice things are likely okay.

If there are reasonable ways to detect this, it would make our lives easier. Otherwise seialization proxy or switching to a custom game format will help with one of our compatibility problems. Switching to something other than RMI woudl help with the other.

**Comment by simon33-2** [comment]
Running both versions in a networked game works testing with two instances looping back on the same PC.

File: src/main/java/games/strategy/triplea/delegate/DependentBattle.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.PlayerID;
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.Unit;
+
+/**
+ * Battle with possible dependencies
+ * Includes MustFightBattle and NonFightingBattle
+ */
+abstract public class DependentBattle extends AbstractBattle {
```
Comment (RoiEXLab) [inline src/main/java/games/strategy/triplea/delegate/DependentBattle.java]: Please change this to `public abstract class` in order to be consequent in terms of naming...

Comment (simon33-2) [inline src/main/java/games/strategy/triplea/delegate/DependentBattle.java]: done

**Pull request merged**